src/gtest/test_foundersreward.cpp

```diff
@@ -0,0 +1,64 @@
+#include <gtest/gtest.h>
+
+#include "chainparams.h"
+#include "utilstrencodings.h"
+#include <string>
+#include <set>
+
+// To run test:
+// ./zcash-gtest --gtest_filter="founders_reward_test.*"
+
+TEST(founders_reward_test, testnet) {
+    SelectParams(CBaseChainParams::TESTNET);
+
+    CChainParams params = Params();
+    
+    // For Testnet, the first address is derived from the old script. 
+    // FOUNDERS_REWARD_SCRIPT = a9146708e6670db0b950dac68031025cc5b63213a49187
+    //                address = 2N2e2FRfP9D1dRN1oRWkH7pbFM69eGNAuQ4
+    EXPECT_EQ(params.GetFoundersRewardScript(1), ParseHex("a9146708e6670db0b950dac68031025cc5b63213a49187"));
+    EXPECT_EQ(params.GetFoundersRewardAddress(1), "2N2e2FRfP9D1dRN1oRWkH7pbFM69eGNAuQ4");
+    size_t nInterval = params.GetConsensus().nSubsidyHalvingInterval;
+    
+    // If the block height parameter is out of bounds, there is an assert.
+    // Block height must be >0 and <nSubsidyHalvingInterval.

```

Comment:
Update this depending on the decision about https://github.com/zcash/zcash/pull/1398/files#r78873403